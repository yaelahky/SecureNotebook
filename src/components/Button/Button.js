import React from 'react';
import {TouchableOpacity, Text, View} from 'react-native';
import styles from './Button.styles';

const Button = props => {
  const {title, onPress, style} = props;
  return (
    <TouchableOpacity
      onPress={onPress}
      activeOpacity={0.5}
      style={[styles.container, {...style}]}>
      <View style={styles.itemContainer}>
        <Text style={styles.text}>{title}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default Button;
