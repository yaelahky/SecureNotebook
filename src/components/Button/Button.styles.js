import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#009043',
    padding: 16,
    borderRadius: 32,
    minWidth: '100%',
  },
  itemContainer: {
    alignItems: 'center',
  },
  text: {
    fontWeight: '500',
    fontSize: 14,
    color: 'white',
  },
});

export default styles;
