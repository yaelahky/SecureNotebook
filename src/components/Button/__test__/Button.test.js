import React from 'react';
import {render} from '@testing-library/react-native';
import Button from '../Button';

describe('Button', () => {
  beforeEach(() => {});

  afterEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  test('should render Button default', () => {
    const {toJSON} = render(<Button />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render Button with title', () => {
    const {toJSON} = render(<Button title="Press Me" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
