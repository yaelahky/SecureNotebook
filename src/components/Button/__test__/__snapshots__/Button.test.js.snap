// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button default 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#009043",
      "borderRadius": 32,
      "minWidth": "100%",
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    />
  </View>
</View>
`;

exports[`Button should render Button with title 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#009043",
      "borderRadius": 32,
      "minWidth": "100%",
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Press Me
    </Text>
  </View>
</View>
`;
